### PR TITLE
Admedia Bid Adapter : initial adapter release

### PR DESCRIPTION
**Pull request by admediadev**
## Type of change

- [x] New bidder adapter   
https://github.com/prebid/prebid.github.io/pull/4346
## Description of change

Contact Email - [developers@admedia.com](mailto:developers@admedia.com)
Test Parameters -
{
bidder: 'admedia',
params: {
placementId: '782332',
aid: '86858',
},
refererInfo: {
page: "http://test.com/"
}
}

## Other information

Doc PR - https://github.com/prebid/prebid.github.io/pull/4346

### CODE REVIEW

**Comment by ChrisHuie** [comment]
@admediadev looks like you have a failing unit test.

**Comment by admediadev** [comment]
Hello @ChrisHuie , the code is updated with revised test case, please go ahead with the review. Thanks.

File: modules/admediaBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    return validBidRequests.map(bidRequest => {
+      let sizes = []
+      if (bidRequest.mediaTypes && bidRequest.mediaTypes[BANNER] && bidRequest.mediaTypes[BANNER].sizes) {
+        sizes = bidRequest.mediaTypes[BANNER].sizes;
+      }
+
+      var tagData = [];
+
+      var tagObjects = [];
+      tagObjects['tags'] = [];
```
Comment (mmoschovas) [inline modules/admediaBidAdapter.js:41]: What is the purpose of this? This doesnt seem to be referenced here and will always be returned as an empty array.

File: modules/admediaBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      for (var i = 0, j = sizes.length; i < j; i++) {
+        let tag = {};
+        tag.sizes = [];
+        tag.id = bidRequest.params.placementId;
+        tag.aid = bidRequest.params.aid;
+        tag.sizes.push(sizes[i].toString().replace(',', 'x'));
+        tagObjects.push(tag);
+      }
+
+      Object.assign(tagData, tagObjects);
```
Comment (mmoschovas) [inline modules/admediaBidAdapter.js:51]: Is there any reason to copy data over from tagObjects to tagData and not just use the original here?

File: modules/admediaBidAdapter.js [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+  /**
+   * Unpack the response from the server into a list of bids.
+   *
+   * @param {ServerResponse} serverResponse A successful response from the server.
+   * @param bidRequest
+   * @return {Bid[]} An array of bids which were nested inside the server.
+   */
+  interpretResponse: function (serverResponse, bidRequest) {
+    const requiredKeys = ['requestId', 'cpm', 'width', 'height', 'ad', 'ttl', 'creativeId', 'netRevenue', 'currency'];
```
Comment (mmoschovas) [inline modules/admediaBidAdapter.js:78]: meta.advertiserDomains is also a required field as per https://docs.prebid.org/dev-docs/bidder-adaptor.html#interpreting-the-response

File: modules/admediaBidAdapter.js [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        for (const requiredKey of requiredKeys) {
+          if (!bid.hasOwnProperty(requiredKey)) {
+            return [];
+          }
+          bidResponse[requiredKey] = bid[requiredKey];
+        }
+        validBidResponses.push(bidResponse);
+      });
+    }
+    // console.log(JSON.stringify(validBidResponses));
```
Comment (mmoschovas) [inline modules/admediaBidAdapter.js:94]: Can this line be droppped please

File: modules/admediaBidAdapter.js (hunk 1 above)
Comment (admediadev) [reply modules/admediaBidAdapter.js]: Thanks @mmoschovas , removed it as not needed.

File: modules/admediaBidAdapter.js (hunk 2 above)
Comment (admediadev) [reply modules/admediaBidAdapter.js]: Right, not needed, using original object now @mmoschovas . Thanks

File: modules/admediaBidAdapter.js (hunk 3 above)
Comment (admediadev) [reply modules/admediaBidAdapter.js]: Added check for meta and advertiserDoamins. Thanks

File: modules/admediaBidAdapter.js (hunk 4 above)
Comment (admediadev) [reply modules/admediaBidAdapter.js]: Dropped.Thanks.

File: test/spec/modules/admediaBidAdapter_spec.js [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+				  'advertiserDomains': [
+			        'https://www.test.com'
+				  ]
+			    }
+			  }
+          ]
+		  }
+      ]
+      let result = spec.interpretResponse(serverResponse, bidRequest[0]);
+      expect(Object.keys(result)).to.deep.equal(Object.keys(expectedResponse));
```
[the remark this answers is not in the document]
Comment (admediadev) [reply test/spec/modules/admediaBidAdapter_spec.js]: Thanks @mmoschovas for pointing out, fixed the test case as well.

**Review by mmoschovas — APPROVED** [review]
Thanks for making these changes!

**Comment by admediadev** [comment]
Thank you @mmoschovas .

**Pull request merged**